### PR TITLE
Move mailer error below tabs [MAILPOET-1820]

### PR DESCRIPTION
**Pull request by amine-mp**
[MAILPOET-1820]

[MAILPOET-1820]: https://mailpoet.atlassian.net/browse/MAILPOET-1820

### CODE REVIEW

File: assets/js/src/newsletters/listings/mixins.jsx [hunk 1]
```diff
@@ -401,134 +401,6 @@ const MailerMixin = {
...
-      MailPoet.Notice.success(MailPoet.I18n.t('mailerSendingResumedNotice'));
-      window.mailpoet_listing.forceUpdate();
-    }).fail((response) => {
-      if (response.errors.length > 0) {
-        MailPoet.Notice.error(
-          response.errors.map(error => error.message),
-          { scroll: true }
-        );
-      }
-    });
```
Comment (JanJakes) [inline assets/js/src/newsletters/listings/mixins.jsx]: So if I understand this correctly:
1. Originally this used to show error notice if error found on `checkMailerStatus` call and then render its body. This is called from multiple places.
2. Now it only shows the notice and expects the body will be rendered by `MailerError` using `{ this.state.meta.mta_method && <MailerError {...this.state.meta} /> }` line.

Is that right? Maybe I just misunderstood.

Now, doesn't this change behavior significantly? To have the body actually rendered it's necessary that `listings.jsx` is rendered? How is that guaranteed?

Also, it would be nice to get rid of mixins but doesn't this make the code actually worse? Rendering notice without body and hoping it will appear there feels really not good.

Comment (amine-mp) [reply assets/js/src/newsletters/listings/mixins.jsx]: You're misunderstanding, let me explain. The `MailerMixin.checkMailerStatus` was doing the following:

1- if there is an authorization error, it shows a notice with id `mailpoet_authorization_error`.
2- if the sending is paused due to an other type of sending errors, it creates a container with id `mailpoet_mailer_error` if not present, then renders some JSX into that container.

I kept the first notice (1) in the Mixin because it does not relate to the task and moved the second one. Now the mixin will still be called to do (1), while the part (2) will be done in pure React (JSX is rendered directly into the component, no container is created).

Does that make sense?

Comment (JanJakes) [reply assets/js/src/newsletters/listings/mixins.jsx]: Right, I was wrong with the body, `mailpoet_authorization_error` is another notice, not a wrapper.

However, the change of behavior is still there - rendering of  `mailpoet_mailer_error` was originally triggered by `checkMailerStatus`, not anymore. This means that it can (maybe) happen that the `mailpoet_mailer_error` will not render when it should - `checkMailerStatus` was originally called from various places. Are *all* of them followed by render of `MailerError` in the current implementation? Just double checking.

File: assets/js/src/listing/notices.jsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  return null;
+}
+
+MailerError.propTypes = {
+  mta_method: PropTypes.string.isRequired,
+  mta_log: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+};
+
+const PHPMailerCheckSettingsNotice = () => (
+  <Fragment>
```
Comment (JanJakes) [inline assets/js/src/listing/notices.jsx]: Why don't you keep `<>`?

File: assets/js/src/listing/notices.jsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      MailPoet.Notice.error(
+        response.errors.map(error => error.message),
+        { scroll: true }
+      );
+    }
+  });
+};
+
+function MailerError(props) {
+  if (props.mta_log.error && props.mta_log.status === 'paused' && props.mta_log.error.operation !== 'authorization') {
```
Comment (JanJakes) [inline assets/js/src/listing/notices.jsx]: Whole function body is wrapped in this `if`. It would be nicer to early-return `null` here.

File: assets/js/src/listing/notices.jsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        </div>
+      );
+    }
+
+    let message = props.mta_log.error.error_message;
+    const code = props.mta_log.error.error_code;
+    if (code) {
+      message += message ? ', ' : '';
+      message += MailPoet.I18n.t('mailerErrorCode').replace('%$1s', code);
+    }
```
Comment (JanJakes) [inline assets/js/src/listing/notices.jsx]: I may look like some logic is just moved elsewhere but it's not. Please, try to split next time code moving and refactoring in different commits. I's extremely hard to see changes when they are hidden in a single commit that just moves huge blocks. You changed some naming and variables/constants here. Don't hide it. Show it explicitly in a separate commit.

File: assets/js/src/listing/notices.jsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    if (response.errors.length > 0) {
+      MailPoet.Notice.error(
+        response.errors.map(error => error.message),
+        { scroll: true }
+      );
+    }
+  });
+};
+
+function MailerError(props) {
```
Comment (JanJakes) [inline assets/js/src/listing/notices.jsx]: Why do you use sometimes `const` and sometimes `function`?

File: assets/js/src/listing/notices.jsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+import React, { Fragment } from 'react';
+import ReactStringReplace from 'react-string-replace';
+
+const resumeMailerSending = () => {
+  MailPoet.Ajax.post({
+    api_version: window.mailpoet_api_version,
+    endpoint: 'mailer',
+    action: 'resumeSending',
+  }).done(() => {
+    MailPoet.Notice.success(MailPoet.I18n.t('mailerSendingResumedNotice'));
```
Comment (JanJakes) [inline assets/js/src/listing/notices.jsx]: Above this line there was a `MailPoet.Notice.hide('mailpoet_mailer_error');`. Why did it dissapear? Maybe there's a reason for that but you removed it in a single commit with a huge move so it's extremely hard to even spot this. And since it's not a separate commit I have no idea if it's an error or not.

File: assets/js/src/listing/notices.jsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+      );
+    }
+  });
+};
+
+function MailerError(props) {
+  if (props.mta_log.error && props.mta_log.status === 'paused' && props.mta_log.error.operation !== 'authorization') {
+    if (props.mta_log.error.operation === 'migration') {
+      return (
+        <div className="mailpoet_notice notice notice-warning">
```
Comment (JanJakes) [inline assets/js/src/listing/notices.jsx]: So here we can't use `MailPoet.Notice` so we have to imitate it? That's really not nice but if there's no other option at the moment, what can we do.

File: assets/js/src/listing/notices.jsx (hunk 2 above)
Comment (amine-mp) [reply assets/js/src/listing/notices.jsx]: My syntax highlighter was not happy with it, so I changed it for the dev time, then forgot to bring it back. I will edit that.

File: assets/js/src/listing/notices.jsx (hunk 3 above)
Comment (amine-mp) [reply assets/js/src/listing/notices.jsx]: You're right.

File: assets/js/src/listing/notices.jsx (hunk 4 above)
Comment (amine-mp) [reply assets/js/src/listing/notices.jsx]: Ok

File: assets/js/src/listing/notices.jsx (hunk 5 above)
Comment (amine-mp) [reply assets/js/src/listing/notices.jsx]: I prefer to use `const`, I will rewrite this.

File: assets/js/src/listing/notices.jsx (hunk 6 above)
Comment (amine-mp) [reply assets/js/src/listing/notices.jsx]: I removed it because we don't use `MailPoet.Notice` to render the `mailpoet_mailer_error` anymore.
The notice will be hidden (not rendered) by React. The next line calls `window.mailpoet_listing.forceUpdate()` which will rerender the `Listing` component which will rerender `MailerError` which will return `null` because the sending was resumed.

File: assets/js/src/listing/notices.jsx (hunk 7 above)
Comment (amine-mp) [reply assets/js/src/listing/notices.jsx]: We are not imitating `MailPoet.Notice` here, I kept the CSS classes just to keep the same display.

File: assets/js/src/newsletters/listings/mixins.jsx (hunk 1 above)
Comment (amine-mp) [reply assets/js/src/newsletters/listings/mixins.jsx]: `checkMailerStatus` is called in the 4 emails listings. they all render the `Listing` component and pass a callback which will call `checkMailerStatus` after loading emails data. With the current implementation, the `Listing` component will render `MailerError` itself whenever it's rendered and there is some sending error in the meta data loaded from the API.
You may say: "But we are using `Listing` on other pages like Forms and Subscribers, we don't want the `MailerError` to render there!". That's correct, but notice that the API only includes `mta_log` and `mta_method` on the meta if we requested newsletters. So `MailerError` will always render `null` on other pages.
